folly/Random.cpp

```diff
@@ -26,16 +26,33 @@
 #include <folly/File.h>
 #include <folly/FileUtil.h>
 
+#ifdef _MSC_VER
+#include <wincrypt.h>
+#endif
+
 namespace folly {
 
 namespace {
 
 void readRandomDevice(void* data, size_t size) {
+#ifdef _MSC_VER
+  static bool initialized = false;
```

Comment:
(From @yfeldblum) This is not threadsafe. Two threads that simultaneously call readRandomDevice can overwrite each other's data. Using std::once_flag should help.

```
#include <mutex>

static std::once_flag flag;
static HCRYPTPROV provider;
std::call_once(flag, [&] }
  PCHECK(CryptAcquireContext(&provider, nullptr, nullptr, PROV_RSA_FULL, 0));
});
CHECK(size <= std::numeric_limits<DWORD>::max()); // not PCHECK
PCHECK(CryptGenRandom(provider, (DWORD)size, (BYTE*)data);
```

The other possibility is to use ThreadLocal.